### DISCUSSION TITLE
java.lang.NoClassDefFoundError: io/fabric8/kubernetes/model/jackson/UnmatchedFieldTypeModule

### DISCUSSION THREAD

**Discussion by alonir** [fabric8io/kubernetes-client#4844] [Q&A] [answered]
Hi,

Upgrading from 5.12.3 to 6.4.1 failing on the below error, did someone encounter it before?

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ocpClient': Invocation of init method failed; nested exception is **java.lang.NoClassDefFoundError: io/fabric8/kubernetes/model/jackson/UnmatchedFieldTypeModule**\n\tat org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:160)\n\tat org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:440)\n\tat org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796)\n\tat org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)\n\tat org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)\n\tat org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)\n\tat org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)\n\tat org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)\n\tat org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)\n\tat org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)\n\tat org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1391)\n\tat org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1311)\n\tat org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)\n\tat org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)\n\t... 42 common frames omitted\nCaused by: java.lang.NoClassDefFoundError: io/fabric8/kubernetes/model/jackson/UnmatchedFieldTypeModul

**Comment by alonir** [comment, marked as answer]
Just saw another answer which helped me needed to add the below dependency to solve the issue:

<dependency>
            <groupId>io.fabric8</groupId>
            <artifactId>kubernetes-model-common</artifactId>
            <version>6.4.1</version>
        </dependency>

**Reply by shawkins** [reply]
See also https://github.com/fabric8io/kubernetes-client/blob/master/doc/FAQ.md#ive-tried-adding-a-dependency-to-kubernetes-client-but-im-still-getting-weird-class-loading-issues-what-gives
